beacon-chain/blockchain/process_block.go

```diff
@@ -582,6 +588,10 @@ func (s *Service) savePostStateInfo(ctx context.Context, r [32]byte, b interface
 	if err := s.cfg.StateGen.SaveState(ctx, r, st); err != nil {
 		return errors.Wrap(err, "could not save state")
 	}
+	// Save state root to block root mapping
+	if err := s.cfg.BeaconDB.SaveBlockRoot(ctx, b.Block().StateRoot(), r); err != nil {
```

Comment:
The block root should be saved in the db with the block rather than be something initiated by a consumer of the db package. So this should be implemented directly in `SaveBlock`